File: backend/users/urls.py
from django.urls import path
from rest_framework_simplejwt.views import TokenRefreshView, TokenVerifyView
from .views import UserRegistrationView, UserLoginView, UserProfileView, UserChangePasswordView, SendPasswordResetEmailView, UserPasswordResetView


urlpatterns = [
    path('token/refresh/', TokenRefreshView.as_view(), name='token-refresh'),
    path('token/verify/', TokenVerifyView.as_view(), name='token-verify'),
    path('register/', UserRegistrationView.as_view(), name='register'),
    path('login/', UserLoginView.as_view(), name='login'),
    path('profile/', UserProfileView.as_view(), name='profile'),
    path('change-password/', UserChangePasswordView.as_view(),
         name='change-password'),
    path('send-password-reset-email/', SendPasswordResetEmailView.as_view(),
         name='send-password-reset-email'),
    path('password-reset/<str:user_id>/<str:token>/',
         UserPasswordResetView.as_view(), name='reset-password'),
]

File: frontend/src/api/urls.ts
const ROOT = "http://127.0.0.1:8000";

const urls = {
  refreshToken: `${ROOT}/token/refresh/`,
  verifyToken: `${ROOT}/token/verify/`,
  register: `${ROOT}/register/`,
  login: `${ROOT}/login/`,
  profile: `${ROOT}/profile/`,
  resetPassword: `${ROOT}/send-password-reset-email/`
};

export default urls;
